In [1]:
import pandas as pd
import dtale


In [2]:
print(pd.__version__)

3.0.5


In [3]:
pedidos_destino = pd.read_csv(r'C:\Users\luizm\Desktop\ProjetoX\TesteEngenhariaDeDados\pedidos_destino.csv')
pedidos_origem = pd.read_csv(r'C:\Users\luizm\Desktop\ProjetoX\TesteEngenhariaDeDados\pedidos_origem.csv')

In [4]:
pedidos_destino.dtypes

id_pedido          int64
id_cliente         int64
status               str
valor_total      float64
atualizado_em        str
dtype: object

In [5]:
pedidos_origem.dtypes

id_pedido          int64
id_cliente         int64
status               str
valor_total      float64
atualizado_em        str
operacao             str
dtype: object

In [6]:
resultado = (
    pedidos_origem.groupby("id_pedido")["status"]
    .count()
    .reset_index(name="qtd_status")
    .sort_values(by="qtd_status", ascending=False) 
)
resultado

,id_pedido,qtd_status
28,10029,8
10,10011,7
105,10106,6
21,10022,5
17,10018,5
...,...,...
122,10123,1
128,10129,1
119,10120,1
145,10146,1


In [7]:
qtd_resultado = resultado.sort_values(by= 'qtd_status', ascending=False)
qtd_resultado1 = resultado[resultado['qtd_status'] != 1]
qtd_resultado1

,id_pedido,qtd_status
28,10029,8
10,10011,7
105,10106,6
21,10022,5
17,10018,5
...,...,...
67,10068,2
82,10083,2
141,10142,2
99,10100,2


In [8]:
qtd_destino = (
    pedidos_destino.groupby("id_pedido")["status"]
    .count()
    .reset_index(name="qtd_status")
    .sort_values(by="qtd_status", ascending=False) 
)


In [9]:
qtd_destino = qtd_destino.sort_values(by= 'qtd_status', ascending=False)
qtd_destino1 = qtd_destino[qtd_destino['qtd_status'] != 1]
qtd_destino1

,id_pedido,qtd_status
1,10002,2
73,10079,2
111,10127,2
30,10032,2


In [10]:
qtd_destino

,id_pedido,qtd_status
1,10002,2
73,10079,2
111,10127,2
30,10032,2
0,10001,1
...,...,...
128,10146,1
129,10147,1
130,10148,1
131,10149,1


In [11]:
dup_origem = (
            pedidos_origem[pedidos_origem.duplicated(keep=False)]
            .sort_values('id_pedido'))
dup_origem
#Duplicado na origem, origem não está correta, putz
#Removi duplicados, pois nao faz sentido manter 2 logs para apenas 1 evento

,id_pedido,id_cliente,status,valor_total,atualizado_em,operacao
84,10006,527,CANCELADO,3838.50,2026-06-26T18:26:00-03:00,U
460,10006,527,CANCELADO,3838.50,2026-06-26T18:26:00-03:00,U
161,10061,556,NOVO,1660.90,2026-06-19T20:12:00-03:00,I
147,10061,556,NOVO,1660.90,2026-06-19T20:12:00-03:00,I
333,10076,508,NOVO,810.57,2026-06-27T11:21:00-03:00,I
78,10076,508,NOVO,810.57,2026-06-27T11:21:00-03:00,I
144,10096,562,NOVO,343.21,2026-06-01T23:05:00-03:00,I
148,10096,562,NOVO,343.21,2026-06-01T23:05:00-03:00,I
248,10108,502,ENTREGUE,3398.35,2026-06-01T14:07:00-03:00,U
168,10108,502,ENTREGUE,3398.35,2026-06-01T14:07:00-03:00,U


In [12]:
#retirar duplicidade
origem = pedidos_origem.drop_duplicates()
#normalizar status
origem['status'] = (origem['status'].fillna('').astype(str).str.strip().str.upper())

#id to string
origem['id_pedido'] = origem['id_pedido'].astype(str)
origem['id_cliente'] = origem['id_cliente'].astype(str)



origem

,id_pedido,id_cliente,status,valor_total,atualizado_em,operacao
0,10107,546,NOVO,609.47,2026-05-31T23:31:00-03:00,I
1,10101,511,ENVIADO,4086.09,2026-06-09T20:20:00-03:00,U
2,10127,535,ENTREGUE,4404.93,2026-06-07T19:39:00-03:00,U
3,10024,561,PAGO,4785.77,2026-07-09T01:49:00-03:00,U
4,10075,529,CANCELADO,3896.37,2026-05-29T15:56:00-03:00,U
...,...,...,...,...,...,...
469,10056,563,NOVO,3637.08,2026-05-30T19:23:00-03:00,I
470,10062,558,NOVO,3857.16,2026-06-29T07:18:00-03:00,I
471,10099,557,CANCELADO,192.46,2026-05-25T21:30:00-03:00,U
472,10127,535,ENVIADO,4688.94,2026-06-08T23:39:00-03:00,U


In [13]:
dup_destino = (
            pedidos_destino[pedidos_destino.duplicated(keep=False)]
            .sort_values('id_pedido'))


dup_destino
#somente um duplicado, entao nao é o problema

,id_pedido,id_cliente,status,valor_total,atualizado_em
30,10002,569,NOVO,1912.59,2026-05-13 00:37:00
95,10002,569,NOVO,1912.59,2026-05-13 00:37:00


In [14]:
destino = pedidos_destino.drop_duplicates()
#removi dups pelo msm motivo

#normalizar status
destino['status'] = (destino['status'].fillna('').astype(str).str.strip().str.upper())

#id to string
destino['id_pedido'] = destino['id_pedido'].astype(str)
destino['id_cliente'] = destino['id_cliente'].astype(str)

destino

,id_pedido,id_cliente,status,valor_total,atualizado_em
0,10058,547,CANCELADO,2321.51,2026-06-07 16:27:00
1,10046,540,NOVO,1866.83,2026-05-24 06:56:00
2,10122,521,PAGO,4733.41,2026-06-04 15:49:00
3,10015,552,PAGO,1017.33,2026-05-14 20:58:00
4,10097,524,PAGO,1159.20,2026-07-03 17:11:00
...,...,...,...,...,...
132,10110,526,CANCELADO,2621.88,2026-06-12 01:47:00
133,10003,551,PAGO,502.78,2026-05-20 22:54:00
134,10129,565,NOVO,1563.47,2026-07-21 00:27:00
135,10073,540,ENTREGUE,185.60,2026-05-22 00:49:00


In [15]:
pedidos = origem.merge(destino, how='outer', on=['id_pedido', 'id_cliente', 'valor_total'], indicator='fg_both')
pedidos


,id_pedido,id_cliente,status_x,valor_total,atualizado_em_x,operacao,status_y,atualizado_em_y,fg_both
0,10001,542,NOVO,2058.99,2026-05-18T01:32:00-03:00,I,NaN,NaN,left_only
1,10001,565,NaN,2058.99,NaN,NaN,NOVO,2026-05-18 04:32:00,right_only
2,10002,569,NOVO,1912.59,2026-05-12T21:37:00-03:00,I,NOVO,2026-05-13 00:37:00,both
3,10003,528,PAGO,502.78,2026-05-20T19:54:00-03:00,U,NaN,NaN,left_only
4,10003,528,PAGO,502.78,2026-05-14T14:54:00-03:00,U,NaN,NaN,left_only
...,...,...,...,...,...,...,...,...,...
481,10149,506,CANCELADO,3649.10,2026-06-25T07:21:00-03:00,U,CANCELADO,2026-06-25 10:21:00,both
482,10149,506,CANCELADO,3706.45,2026-06-15T12:21:00-03:00,U,NaN,NaN,left_only
483,10149,506,NOVO,3706.45,2026-06-09T15:21:00-03:00,I,NaN,NaN,left_only
484,10150,570,PAGO,1964.62,2026-07-05T18:21:00-03:00,U,PAGO,2026-07-05 21:21:00,both


In [16]:
pedidos['fg_both'].value_counts()
#left_only = origem 259;  both = ambas
#10 right_only

fg_both
both          259
left_only     217
right_only     10
Name: count, dtype: int64

In [17]:
pedidos_apenas_origem = pedidos[pedidos['fg_both'] == 'left_only']
pedidos_apenas_origem

,id_pedido,id_cliente,status_x,valor_total,atualizado_em_x,operacao,status_y,atualizado_em_y,fg_both
0,10001,542,NOVO,2058.99,2026-05-18T01:32:00-03:00,I,NaN,NaN,left_only
3,10003,528,PAGO,502.78,2026-05-20T19:54:00-03:00,U,NaN,NaN,left_only
4,10003,528,PAGO,502.78,2026-05-14T14:54:00-03:00,U,NaN,NaN,left_only
5,10003,528,NOVO,506.88,2026-05-07T17:54:00-03:00,I,NaN,NaN,left_only
6,10003,528,PAGO,526.02,2026-05-08T12:54:00-03:00,U,NaN,NaN,left_only
...,...,...,...,...,...,...,...,...,...
470,10145,543,ENVIADO,2850.96,2026-05-14T11:17:00-03:00,U,NaN,NaN,left_only
479,10149,506,PAGO,3378.80,2026-06-25T07:21:00-03:00,U,NaN,NaN,left_only
480,10149,506,CANCELADO,3378.80,2026-06-18T18:21:00-03:00,U,NaN,NaN,left_only
482,10149,506,CANCELADO,3706.45,2026-06-15T12:21:00-03:00,U,NaN,NaN,left_only


In [18]:
pedidos_both = pedidos[pedidos['fg_both'] == 'both']
pedidos_both

,id_pedido,id_cliente,status_x,valor_total,atualizado_em_x,operacao,status_y,atualizado_em_y,fg_both
2,10002,569,NOVO,1912.59,2026-05-12T21:37:00-03:00,I,NOVO,2026-05-13 00:37:00,both
10,10004,517,ENVIADO,4079.16,2026-07-04T14:03:00-03:00,U,ENVIADO,2026-07-04 17:03:00,both
11,10005,525,NOVO,564.59,2026-06-07T05:30:00-03:00,I,NOVO,2026-06-07 08:30:00,both
14,10006,527,ENVIADO,3838.50,2026-06-25T20:26:00-03:00,U,CANCELADO,2026-06-26 21:26:00,both
15,10006,527,CANCELADO,3838.50,2026-06-26T18:26:00-03:00,U,CANCELADO,2026-06-26 21:26:00,both
...,...,...,...,...,...,...,...,...,...
477,10147,538,ENTREGUE,4234.36,2026-06-09T06:53:00-03:00,U,PAGO,2026-06-21 21:53:00,both
478,10148,541,NOVO,716.04,2026-05-31T13:39:00-03:00,I,NOVO,2026-05-31 16:39:00,both
481,10149,506,CANCELADO,3649.10,2026-06-25T07:21:00-03:00,U,CANCELADO,2026-06-25 10:21:00,both
484,10150,570,PAGO,1964.62,2026-07-05T18:21:00-03:00,U,PAGO,2026-07-05 21:21:00,both


In [19]:
pedidos_apenas_destino = pedidos[pedidos['fg_both'] == 'right_only']
pedidos_apenas_destino
#pedidos apenas na tabela destino ??

,id_pedido,id_cliente,status_x,valor_total,atualizado_em_x,operacao,status_y,atualizado_em_y,fg_both
1,10001,565,NaN,2058.99,NaN,NaN,NOVO,2026-05-18 04:32:00,right_only
7,10003,551,NaN,502.78,NaN,NaN,PAGO,2026-05-20 22:54:00,right_only
56,10020,504,NaN,2447.98,NaN,NaN,ENTREGUE,2026-06-06 08:23:00,right_only
99,10031,518,NaN,1178.42,NaN,NaN,NOVO,2026-06-16 00:47:00,right_only
137,10045,543,NaN,1700.77,NaN,NaN,NOVO,2026-06-23 06:06:00,right_only
189,10064,521,NaN,0.00,NaN,NaN,PAGO,2026-05-23 18:12:00,right_only
304,10098,542,NaN,0.00,NaN,NaN,ENTREGUE,2026-06-01 16:11:00,right_only
383,10119,542,NaN,0.00,NaN,NaN,PAGO,2026-07-05 07:06:00,right_only
456,10140,533,NaN,1548.25,NaN,NaN,PAGO,2026-07-14 14:57:00,right_only
459,10141,539,NaN,2840.78,NaN,NaN,PAGO,2026-05-28 03:09:00,right_only


##left_anti##
pedidos_left_anti = origem.merge(destino, how='left_anti', on=['id_pedido', 'id_cliente', 'valor_total'], indicator='fg_ambas')
pedidos_left_anti

destino_left_anti = destino.merge(origem, how='left_anti', on=['id_pedido', 'id_cliente', 'valor_total'], indicator='fg_both')
destino_left_anti

pedidos2 = pedidos_destino.merge(pedidos_origem, how='left', on=['id_pedido'])
pedidos2                                

In [20]:
soma_destinos = (pedidos_apenas_destino['status_x']!= 'CANCELADO' ) & (pedidos_apenas_destino['status_y']!= 'CANCELADO' ) 
soma_destinos_final = pedidos_apenas_destino.loc[soma_destinos,'valor_total'].sum()
soma_destinos_final

np.float64(12277.970000000001)

In [21]:
soma_origem = (pedidos_apenas_origem['status_x']!= 'CANCELADO' ) & (pedidos_apenas_origem['status_y']!= 'CANCELADO' ) 
soma_origem_final = pedidos_apenas_origem.loc[soma_origem,'valor_total'].sum()
soma_origem_final

np.float64(473490.93)

In [22]:
soma_pedidos= (pedidos['status_x']!= 'CANCELADO' ) & (pedidos['status_y']!= 'CANCELADO' ) 
soma_pedidos_final = pedidos.loc[soma_pedidos,'valor_total'].sum()
soma_pedidos_final

np.float64(888804.1499999999)

In [23]:
pedidos['status_x'].unique()

<StringArray>
['NOVO', nan, 'PAGO', 'ENTREGUE', 'ENVIADO', 'CANCELADO']
Length: 6, dtype: str

In [24]:
valor_origem = origem.sort_values(by='atualizado_em',ascending=False)
valor_origem

,id_pedido,id_cliente,status,valor_total,atualizado_em,operacao
343,10029,509,PAGO,2165.18,2026-07-19T16:17:00-03:00,U
110,10029,509,NOVO,2165.18,2026-07-18T10:17:00-03:00,I
73,10140,510,PAGO,1548.25,2026-07-14T11:57:00-03:00,U
183,10024,561,PAGO,4785.77,2026-07-12T17:49:00-03:00,D
379,10092,551,CANCELADO,976.86,2026-07-12T13:47:00-03:00,U
...,...,...,...,...,...,...
398,10115,568,ENTREGUE,4384.05,2026-05-07T10:56:00-03:00,U
211,10026,528,NOVO,1340.47,2026-05-06T21:09:00-03:00,I
93,10115,568,NOVO,4384.05,2026-05-04T21:56:00-03:00,I
334,10130,552,NOVO,3798.25,2026-05-04T20:03:00-03:00,I


In [25]:
real_destino_pedido = valor_origem.drop_duplicates(subset=['id_pedido'], keep='first')
real_destino_pedido

,id_pedido,id_cliente,status,valor_total,atualizado_em,operacao
343,10029,509,PAGO,2165.18,2026-07-19T16:17:00-03:00,U
73,10140,510,PAGO,1548.25,2026-07-14T11:57:00-03:00,U
183,10024,561,PAGO,4785.77,2026-07-12T17:49:00-03:00,D
379,10092,551,CANCELADO,976.86,2026-07-12T13:47:00-03:00,U
358,10027,570,PAGO,4520.63,2026-07-10T15:20:00-03:00,U
...,...,...,...,...,...,...
277,10059,543,CANCELADO,1768.66,2026-05-11T22:48:00-03:00,U
195,10044,538,ENTREGUE,2383.75,2026-05-11T08:49:00-03:00,U
394,10037,529,NOVO,1403.87,2026-05-10T09:32:00-03:00,I
221,10088,514,PAGO,2460.68,2026-05-09T13:48:00-03:00,U


valor_destino = destino.sort_values(by='atualizado_em',ascending=False)
valor_destino2 = valor_destino.drop_duplicates(subset=['id_pedido'], keep='first')


valor_destino3 = valor_destino2[
    (~valor_destino2['status'].isin(['CANCELADO', 'cancelado'])) |
    (~valor_destino2['id_pedido'].isin([10110, 10092, 10087]))
].copy()
valor_destino_cancelado = valor_destino2[
    (valor_destino2['status'].isin(['CANCELADO', 'cancelado'])) |
    (valor_destino2['id_pedido'].isin([10110, 10092, 10087]))
].copy()
#valor_destino_cancelado = valor_destino2[valor_destino2['status']=='CANCELADO']

In [26]:
valor_destino = destino.sort_values(by='atualizado_em', ascending=False)
valor_destino2 = valor_destino.drop_duplicates(subset=['id_pedido'], keep='first').copy()

#normaliza status
valor_destino2['status'] = (
    valor_destino2['status']
    .fillna('')
    .astype(str)
    .str.strip()
    .str.upper()
)

valor_destino2['id_pedido'] = valor_destino2['id_pedido'].astype(int)

cancelado = valor_destino2['status'].eq('CANCELADO')

valor_destino3 = valor_destino2[~(cancelado)].copy()



valor_destino_cancelado = valor_destino2[cancelado].copy()

In [27]:
valor_destino_cancelado

,id_pedido,id_cliente,status,valor_total,atualizado_em
108,10092,551,CANCELADO,976.86,2026-07-12 16:47:00
113,10134,546,CANCELADO,359.51,2026-07-06 11:47:00
94,10063,533,CANCELADO,1382.35,2026-07-06 04:06:00
51,10082,517,CANCELADO,994.27,2026-06-29 15:42:00
6,10071,568,CANCELADO,1038.30,2026-06-29 03:20:00
115,10006,527,CANCELADO,3838.50,2026-06-26 21:26:00
40,10149,506,CANCELADO,3649.10,2026-06-25 10:21:00
102,10081,541,CANCELADO,1569.46,2026-06-15 00:07:00
78,10087,525,CANCELADO,400.64,2026-06-12 19:53:00
132,10110,526,CANCELADO,2621.88,2026-06-12 01:47:00


In [28]:
d=dtale.show(valor_destino3)
d

In [29]:
valor_final_destino_cancelado = valor_destino_cancelado['valor_total'].sum()
valor_final_destino_cancelado
#27%

np.float64(63570.75)

In [30]:
valor_final_destino = valor_destino3['valor_total'].sum()
valor_final_destino


np.float64(232495.52999999997)

comp_destino['delta_status'] = comp_destino['status_x'].eq(comp_destino['status_y']).where(
    comp_destino['status_x'].eq(comp_destino['status_y']),'DIFERENTE').fillna('IGUAL')


In [31]:
comp_destino = real_destino_pedido.merge(destino, 
                                         on='id_pedido', how='outer', indicator='fg_both'
                                            )
comp_destino['d_valor_total'] =  comp_destino['valor_total_x'] - comp_destino['valor_total_y']


comp_destino['delta_status'] = comp_destino['status_x'].eq(comp_destino['status_y']).map({True: 'IGUAL', False: 'DIFERENTE'})



comp_destino

,id_pedido,id_cliente_x,status_x,valor_total_x,atualizado_em_x,operacao,id_cliente_y,status_y,valor_total_y,atualizado_em_y,fg_both,d_valor_total,delta_status
0,10001,542,NOVO,2058.99,2026-05-18T01:32:00-03:00,I,565,NOVO,2058.99,2026-05-18 04:32:00,both,0.0,IGUAL
1,10002,569,NOVO,1912.59,2026-05-12T21:37:00-03:00,I,569,NOVO,1912.59,2026-05-13 00:37:00,both,0.0,IGUAL
2,10003,528,PAGO,502.78,2026-05-20T19:54:00-03:00,U,551,PAGO,502.78,2026-05-20 22:54:00,both,0.0,IGUAL
3,10004,517,ENVIADO,4079.16,2026-07-04T14:03:00-03:00,U,517,ENVIADO,4079.16,2026-07-04 17:03:00,both,0.0,IGUAL
4,10005,525,NOVO,564.59,2026-06-07T05:30:00-03:00,I,525,NOVO,564.59,2026-06-07 08:30:00,both,0.0,IGUAL
...,...,...,...,...,...,...,...,...,...,...,...,...,...
148,10146,544,NOVO,1339.15,2026-06-11T18:43:00-03:00,I,544,NOVO,1339.15,2026-06-11 21:43:00,both,0.0,IGUAL
149,10147,538,PAGO,4234.36,2026-06-21T18:53:00-03:00,U,538,PAGO,4234.36,2026-06-21 21:53:00,both,0.0,IGUAL
150,10148,541,NOVO,716.04,2026-05-31T13:39:00-03:00,I,541,NOVO,716.04,2026-05-31 16:39:00,both,0.0,IGUAL
151,10149,506,PAGO,3378.80,2026-06-25T07:21:00-03:00,U,506,CANCELADO,3649.10,2026-06-25 10:21:00,both,-270.3,DIFERENTE


In [32]:
comp_destino_status_dif = comp_destino[comp_destino['delta_status']=='DIFERENTE']
comp_destino_status_dif

,id_pedido,id_cliente_x,status_x,valor_total_x,atualizado_em_x,operacao,id_cliente_y,status_y,valor_total_y,atualizado_em_y,fg_both,d_valor_total,delta_status
21,10022,526,CANCELADO,1098.82,2026-06-29T23:47:00-03:00,U,526,ENVIADO,1098.82,2026-06-24 22:47:00,both,0.00,DIFERENTE
27,10028,542,PAGO,2496.64,2026-06-29T21:21:00-03:00,U,NaN,NaN,NaN,NaN,left_only,NaN,DIFERENTE
31,10032,563,CANCELADO,5049.10,2026-05-20T22:37:00-03:00,U,563,NOVO,5049.10,2026-05-21 01:37:00,both,0.00,DIFERENTE
50,10050,568,PAGO,3674.19,2026-05-30T08:09:00-03:00,U,568,CANCELADO,3563.92,2026-05-25 00:09:00,both,110.27,DIFERENTE
54,10054,527,ENVIADO,3792.64,2026-06-05T09:03:00-03:00,U,527,PAGO,4096.05,2026-06-05 12:03:00,both,-303.41,DIFERENTE
61,10061,556,PAGO,1660.90,2026-07-02T19:12:00-03:00,D,NaN,NaN,NaN,NaN,left_only,NaN,DIFERENTE
69,10069,559,NOVO,2846.29,2026-05-12T22:41:00-03:00,D,NaN,NaN,NaN,NaN,left_only,NaN,DIFERENTE
72,10072,541,ENVIADO,4293.04,2026-07-09T12:05:00-03:00,D,NaN,NaN,NaN,NaN,left_only,NaN,DIFERENTE
77,10077,511,CANCELADO,1729.44,2026-05-30T04:49:00-03:00,D,NaN,NaN,NaN,NaN,left_only,NaN,DIFERENTE
79,10079,540,ENVIADO,516.00,2026-06-17T08:41:00-03:00,U,540,NOVO,516.00,2026-06-17 11:41:00,both,0.00,DIFERENTE


In [33]:
d=dtale.show(comp_destino_status_dif)
d

In [34]:
comp_destino.dtypes

id_pedido               str
id_cliente_x            str
status_x                str
valor_total_x       float64
atualizado_em_x         str
operacao                str
id_cliente_y            str
status_y                str
valor_total_y       float64
atualizado_em_y         str
fg_both            category
d_valor_total       float64
delta_status            str
dtype: object

In [35]:
comp_destino['fg_both'].value_counts()
#left_only = log origem

fg_both
both          136
left_only      17
right_only      0
Name: count, dtype: int64

In [36]:
comp_destino

,id_pedido,id_cliente_x,status_x,valor_total_x,atualizado_em_x,operacao,id_cliente_y,status_y,valor_total_y,atualizado_em_y,fg_both,d_valor_total,delta_status
0,10001,542,NOVO,2058.99,2026-05-18T01:32:00-03:00,I,565,NOVO,2058.99,2026-05-18 04:32:00,both,0.0,IGUAL
1,10002,569,NOVO,1912.59,2026-05-12T21:37:00-03:00,I,569,NOVO,1912.59,2026-05-13 00:37:00,both,0.0,IGUAL
2,10003,528,PAGO,502.78,2026-05-20T19:54:00-03:00,U,551,PAGO,502.78,2026-05-20 22:54:00,both,0.0,IGUAL
3,10004,517,ENVIADO,4079.16,2026-07-04T14:03:00-03:00,U,517,ENVIADO,4079.16,2026-07-04 17:03:00,both,0.0,IGUAL
4,10005,525,NOVO,564.59,2026-06-07T05:30:00-03:00,I,525,NOVO,564.59,2026-06-07 08:30:00,both,0.0,IGUAL
...,...,...,...,...,...,...,...,...,...,...,...,...,...
148,10146,544,NOVO,1339.15,2026-06-11T18:43:00-03:00,I,544,NOVO,1339.15,2026-06-11 21:43:00,both,0.0,IGUAL
149,10147,538,PAGO,4234.36,2026-06-21T18:53:00-03:00,U,538,PAGO,4234.36,2026-06-21 21:53:00,both,0.0,IGUAL
150,10148,541,NOVO,716.04,2026-05-31T13:39:00-03:00,I,541,NOVO,716.04,2026-05-31 16:39:00,both,0.0,IGUAL
151,10149,506,PAGO,3378.80,2026-06-25T07:21:00-03:00,U,506,CANCELADO,3649.10,2026-06-25 10:21:00,both,-270.3,DIFERENTE


In [37]:
comp_destino2 = comp_destino[
    (comp_destino['delta_status'] != 'IGUAL') |
    (comp_destino['fg_both'] != 'both')].copy()

In [38]:
comp_destino2.to_csv('relatorio_divergencias.csv',index=False)

In [39]:
pedidos_apenas_destino.to_csv('Pedidos_Apenas_Destino.csv', index=False)

2026-09-18 13:14:08,183 - INFO     - Executing shutdown due to inactivity...
2026-09-18 13:14:20,368 - INFO     - Executing shutdown...
2026-09-18 13:14:20,371 - INFO     - Not running with the Werkzeug Server, exiting by searching gc for BaseWSGIServer
